build-system/tasks/make-extension/index.js

```diff
@@ -258,32 +266,91 @@ async function makeExtensionFromTemplates(
 
   log(`${blurb.join('\n\n')}\n`);
 
-  return true;
+  return writtenFiles;
+}
+
+/**
+ * @param {Array<string>} paths
+ * @param {function(...*):*} fn
+ * @return {Promise}
+ */
+async function affectsExistingPaths(paths, fn) {
+  const stashStdout = getStdout(`git stash push --keep-index`);
+
+  await fn();
+
+  const head = getStdout('git rev-parse HEAD').trim();
+  getOutput(`git checkout ${head} ${paths.join(' ')}`);
+
+  if (!stashStdout.startsWith('No local changes')) {
+    getOutput('git stash pop');
+  }
+}
+
+/**
+ * Generates and tests an extension.
+ * (Unit tests for the generator are located in test/test.js)
+ *
+ * @return {!Promise}
+ */
+async function runMakeExtensionTests() {
+  let error;
+
+  await affectsExistingPaths([extensionBundlesJson], async () => {
+    const name = 'amp-generated-for-test';
+
+    const writtenFiles = [
+      ...(await makeExtensionFromTemplates(TEMPLATES_CLASSIC, '.', {
+        name,
+        version: '0.1',
+      })),
+      ...(await makeExtensionFromTemplates(TEMPLATES_BENTO, '.', {
+        name,
+        version: '1.0',
+        bento: true,
+      })),
+    ];
+
+    for (const command of [
+      `amp build --extensions=${name} --core_runtime_only`,
+      `amp unit --nohelp --headless --files="extensions/${name}/**/test/test-*.js"`,
```

Comment:
The logs generated by these commands are very verbose, and I fear they will make the overall logs of the `Checks` job hard to read.

![image](https://user-images.githubusercontent.com/26553114/116271852-604ed980-a74e-11eb-85d1-d74234ddb932.png)

Two suggestions to make the logs less noisy in the passing case:
- Reduce the logging level of `amp make-extension` by replacing `log()` on line 116 and line 267 to `logLocalDev()`
- Instead of `execOrThrow()` on line 320, use `getOutput()` and assign its return value to `result`. Then, check the value of `result.status`. If it's non-zero, log `result.stderr` to the console.